### DISCUSSION TITLE
Signup auth hook

### DISCUSSION THREAD

**Discussion by TheCodeDestroyer** [supabase/supabase#28317] [Feature Requests]
Hiya Supabase! 👋 

From my understanding the new auth hook (beta) functionality is meant as an alternative to the trigger on auth.users table.
It would be really awesome if we could hook into user signup (user insert) hook to run some custom logic
and if there is an error (manually raised in the transaction) properly propagate that error back to the API or client.

**Comment by ggomarighetti** [comment]
Not only sign-up but also sign-in, verification, recovery, and other authentication flows are essential. With Edge Functions now available and everything accessible via HTTP, implementing a solution similar to Ory Actions [1] could be a game-changer. It would allow complete customization of the application's authentication/authorization flow, transforming this project into the ultimate CIAM solution.

[1] [Ory Kratos Hooks Configuration](https://www.ory.sh/docs/kratos/hooks/configure-hooks)
